### ISSUE TITLE
Login: "Your session has expired" notice shown for Google login despite being logged in successfully

### ISSUE THREAD

**Issue by designsimply** [wordpress-mobile/WordPress-iOS#13806]
Steps to reproduce:

1. Tap "Log in".
1. Tap "Continue with Google".
1. Get the verification code from your authenticator app.
1. Enter the verification code into the app.
1. Tap "Next".

**Result:** I often get a session expired notice even though I am logged in successfully. When I tap "OK" then I am taken to the app as logged in. ([1m18s](https://cloudup.com/c7dNWqjiVwv))

<blockquote>Your session has expired, please go back to the login screen.</blockquote>

<img src="https://user-images.githubusercontent.com/1119271/78308513-e970ee00-7505-11ea-8c86-628238d8b68e.png" width="200" alt="your-session-has-expired-1615">
<sup>Tested with WPiOS 14.5.0.1 TestFlight beta on iPhone 11 iOS 13.4.</sup>

**Comment by mindgraffiti** [comment]
Hi @designsimply! 

A quick search for the error message leads me to believe that it's an API error from Google. The next time you see this, would you copy & paste your activity log? It would also be helpful to know if this happened on March 26, 2020 (the day that google had several services go down at the same time) or if it's consistent for you.

I've added it to our UL&S backlog for further investigating.

**Comment by designsimply** [comment]
> A quick search for the error message leads me to believe that it's an API error from Google.

Yes, but should I see an API error in the app if the login goes through normally? It tells me to go back and log in but I am already logged in at that point—also, I'm entering the code in fairly quickly so I don't think the sessions should have expired yet!

Video: [1m49s](https://cloudup.com/cjqlo5hhM4J)
App logs: [session-expired-notice-for-successful-2fa-google-sign-in-wpios-14.6-beta-13806.txt](https://github.com/wordpress-mobile/WordPress-iOS/files/4463512/app-logs-after-session-expired-notice-for-successful-2fa-google-sign-in-wpios-14.6-beta-13806.txt)
<p style="clear:both;"><sup>Tested with WPiOS 14.6.0.1 TestFlight beta on iPhone 6S iOS 13.3.1.</sup></p>

`2020-04-10 15:16:09:143 🔵 Tracked: login_social_button_failure <error: The user canceled the sign-in flow., source: google>`

> It would also be helpful to know if this happened on March 26, 2020

Nope, and this issue happens for me fairly often. I've tested login many many times today and I've seen the session expired error for successful Google sign in flows several times.

**Comment by designsimply** [comment]
I think I figured it out! I think it happens if you tap "Next" more than once on the screen where you enter the verification code. 

Video: [2m18s](https://cloudup.com/cfst9wNyyNl)
App logs: [app-logs-with-and-without-expired-session-notice-for-successful-login.txt](https://github.com/wordpress-mobile/WordPress-iOS/files/4463605/app-logs-with-and-without-expired-session-notice-for-successful-login.txt)
<p style="clear:both;"><sup>Tested with WP Internal 14.6.0.20200410 MS App Center alpha on iPhone 6S iOS 13.3.1.</sup></p>

**Comment by mindgraffiti** [comment]
Activity Log snippet

```
WordPress[84734:7494388] 🔵 Tracked: login_failed_to_login <error_code: 0, error_description: Error Domain=WordPressComOAuthError Code=0 "two_step_nonce required." UserInfo={NSLocalizedDescription=two_step_nonce required., WordPressComOAuthErrorResponseObjectKey={
                code = "empty_two_step_nonce";
                message = "two_step_nonce required.";
2020-05-11 08:27:35:079 WordPress[84734:7494388] two_step_nonce required.
                code = "invalid_two_step_nonce";
                code = "invalid_two_step_nonce";
                code = "invalid_two_step_nonce";
```

**Comment by mindgraffiti** [comment]
**Notes dump**

If I've never signed up for WP using a Google account, and then I go to the sign-in, it will silently start the sign up process and ask me for a 2fa code. BUT the WP.com account 2fa won't work. It'll error out and post "invalid_two_step_nonce" as the alert message. The 2fa code is TEXTED to you. 

I had previously quickly logged into my WP.com email account, then logged out and tried to quickly log in with my Google account. The 2fa codes were entered fairly quickly, and the pop up about "you session has expired" appeared before I even entered a 2fa for Google.

**Comment by designsimply** [comment]
> The 2fa codes were entered fairly quickly, and the pop up about "you session has expired" appeared before I even entered a 2fa for Google.

🤔 possibly a separate issue as well?? See https://github.com/wordpress-mobile/WordPress-iOS/issues/13807#issuecomment-612236448. It seems you can't do these kinds of logins too closely together.

**Comment by mindgraffiti** [comment]
Need to research this next week and verify it is still a problem. I think we fixed this issue because all the "pink" buttons now use the NUX button that blocks users from tapping fast and submitting twice

**Comment by mindgraffiti** [comment]
Moving to Groundskeeping as the UL&S project is wrapping up. While I didn't encounter this error during UL&S work, I'm leaving it open because I can't confirm that it was fixed.

**Comment by jkmassel** [comment]
This is no longer applicable now that we use the WP.com OAuth flow.
